### PR TITLE
Typehint macros

### PR DESCRIPTION
**Pull request by mabar**
- new feature
- BC break? no
- doc PR: let's talk about implementation first

Added support for `{varType string $foo}` and `{templateType TemplateClass}`

TODOs
- `@property` and `@property-read` support for `{templateType}`

Refs
- [typed templates (czech only)](https://phpfashion.com/phpstorm-a-napovidani-nad-this-template)
- [phpstorm latte plugin implementation](https://github.com/mesour/intellij-latte/tree/mn-php-content-and-classes)

cc @mesour

### CODE REVIEW

File: tests/Latte/CoreMacros.templateType.php74.phpt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+if (PHP_VERSION_ID < 70400) {
+	Environment::skip('Typed properties cannot be tested with PHP < 7.4');
+}
+
+$compiler = new Latte\Compiler;
+CoreMacros::install($compiler);
+
+test(function () use ($compiler) {
+	require FileMock::create('
+	class TemplateTypeClass
```
Comment (mabar) [inline tests/Latte/CoreMacros.templateType.php74.phpt]: Not sure if it's possible to define class conditionally. Test fails for php < 7.4 before Environment::skip() is called

**Comment by dg** [comment]
I have added macros `{varType}` and `{templateType}` that do nothing at all: https://github.com/nette/latte/commit/4c5f3fa69f33ff1a8027d574a7dd27b2cb566ef7

**Comment by mabar** [comment]
Good enough for plugin support 👍

**Comment by dg** [comment]
Annotations `@var` generating will be more difficult because it must be done in individual methods.

**Comment by mabar** [comment]
Yeah, I know. First class support for static analysis will be way more difficult, it was more like a concept

**Pull request closed without merging**